components/ai_chat/core/browser/utils.cc

```diff
@@ -178,6 +183,65 @@ const std::string& GetActionTypeQuestion(mojom::ActionType action_type) {
   return iter->second;
 }
 
+uint32_t GetMaxPageContentLength(std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> result =
+      context_size.value_or(kCustomModelDefaultContextSize) *
+      kDefaultCharsPerToken;
+  if (!result.IsValid()) {
+    return std::numeric_limits<uint32_t>::max();
+  }
+  return result.ValueOrDie();
+}
+
+uint32_t GetMaxPageContentLength(const mojom::Model& model) {
+  // For Leo models, return the max_page_content_length directly.
+  if (model.options->is_leo_model_options()) {
+    return model.options->get_leo_model_options()->max_page_content_length;
+  }
+
+  // For custom models, delegate to the context size version.
+  return GetMaxPageContentLength(
+      model.options->get_custom_model_options()->context_size);
+}
+
+uint32_t GetLongConversationWarningLength(
+    std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> checked_context_size =
+      context_size.value_or(kCustomModelDefaultContextSize);
```

Comment:
They are vestiges right now, but I'm not sure if I want to remove them. The Web UI context needs to know if a conversation is getting too long. Leo Models have hard-coded values for `max_page_content_length` and `long_conversation_warning_character_limit`. Custom models have a `context_length`, and from that we need to deduce the equivalent values. I've explored doing this a couple ways, one of which was to add the properties to the custom model objects immediately upon reading them out of the preferences. Another route was to simply pass `kDefaultCharsPerToken` to the Web UI context so that it can, on the spot, decide what the appropriate constraints are for custom models. Neither approach feels good, which is why there is only a sparse solution in place, and these unused (for now) methods.